presto-in-memory/src/main/java/com/facebook/presto/plugin/inmemory/InMemoryMetadata.java

```
@@ -0,0 +1,193 @@
+    {
+        InMemoryTableHandle oldTableHandle = checkType(tableHandle, InMemoryTableHandle.class, "tableHandle");
+        InMemoryTableHandle newTableHandle = new InMemoryTableHandle(
+                oldTableHandle.getConnectorId(),
+                oldTableHandle.getSchemaName(),
+                newTableName.getTableName(),
+                oldTableHandle.getColumnHandles()
+        );
+        tables.remove(oldTableHandle.getTableName());
+        tables.put(newTableName.getTableName(), newTableHandle);
+    }
+
+    @Override
+    public void createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    {
+        ConnectorOutputTableHandle outputTableHandle = beginCreateTable(session, tableMetadata, Optional.empty());
+        finishCreateTable(session, outputTableHandle, ImmutableList.of());
+    }
+
+    @Override
+    public ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, Optional<ConnectorNewTableLayout> layout)
+    {
+        return new InMemoryOutputTableHandle(new InMemoryTableHandle(connectorId, tableMetadata));
+    }
+
+    @Override
+    public void finishCreateTable(ConnectorSession session, ConnectorOutputTableHandle tableHandle, Collection<Slice> fragments)
+    {
+        InMemoryOutputTableHandle inMemoryOutputTableHandle = checkType(tableHandle, InMemoryOutputTableHandle.class, "tableHandle");
+        InMemoryTableHandle table = inMemoryOutputTableHandle.getTable();
+        tables.put(table.getTableName(), table);
+    }
+
+    @Override
+    public ConnectorInsertTableHandle beginInsert(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        return IN_MEMORY_INSERT_TABLE_HANDLE;
+    }
+
+    @Override
+    public void finishInsert(ConnectorSession session, ConnectorInsertTableHandle insertHandle, Collection<Slice> fragments) {}
```

it should raise an error that insert is not supported